packages/components/index.d.ts

```diff
@@ -0,0 +1,335 @@
+import * as React from 'react';
```

Comment:
![image](https://user-images.githubusercontent.com/15332326/76565764-9c23d280-64ab-11ea-994b-8b2a3e28520a.png)

:point_up: this is why people start using ReasonML. `SxProps` and `SxProps` are not identical. Cool.

The error is on my side.
1. I've added `@types/theme-ui__components` to `theme-ui` deps to export types from the root package, because adopting TS in `@theme-ui/components` is a lot more work.
2. `@types/theme-ui__components` imported `SxStyleProp` from `@types/theme-ui`.
3. We end up with four lines with `extends SxProps` instead of two, as we want to.

![image](https://user-images.githubusercontent.com/15332326/76566221-98448000-64ac-11ea-891e-6ebd8c43d4f9.png)

---

Hey @kripod! I copied `@types/theme-ui_components` here.
When v0.4 is released with TypeScript support we can get rid of the package from DefinitelyTyped (i.e.change it to a stub logging "@theme-ui/components provides its own definitions").